### PR TITLE
Use Deno CLI as a library instead of shelling out to bundle

### PR DESCRIPTION
**Pull request by shadaj**
Use Deno CLI as a library instead of shelling out to bundle

### CODE REVIEW

File: crates/deno-subsystem/deno-model-builder/Cargo.toml [hunk 1]
```diff
@@ -10,6 +10,12 @@ bincode.workspace = true
 core-plugin-interface = { path = "../../core-subsystem/core-plugin-interface" }
 subsystem-model-builder-util = { path = "../../subsystem-util/subsystem-model-builder-util" }
 deno-model = { path = "../deno-model" }
+deno_emit = "0.16.0"
+url = "2.3.1"
+tokio.workspace = true
+anyhow.workspace = true
+deno = { git = "https://github.com/exograph/deno.git", branch = "patched" }
```
Comment (ramnivas) [inline crates/deno-subsystem/deno-model-builder/Cargo.toml:17]: Please add a comment on the purpose served by the patched version.

File: crates/deno-subsystem/deno-model-builder/src/system_builder.rs [hunk 2]
```diff
@@ -86,23 +89,51 @@ fn process_script(
     module_fs_path: &Path,
 ) -> Result<Vec<u8>, ModelBuildingError> {
     module_skeleton_generator::generate_module_skeleton(module, base_system, module_fs_path)?;
-
```
Comment (ramnivas) [inline crates/deno-subsystem/deno-model-builder/src/system_builder.rs:89]: Remove the Deno prerequisite from the top-level README.

**Review by ramnivas — APPROVED** [review]
Added a few minor issues, but otherwise awesome!

File: Cargo.toml [hunk 3]
```diff
@@ -50,10 +50,10 @@ async-recursion = "1.0.0"
 async-trait = "0.1.53"
 bincode = "1.3.3"
 bytes = "1"
-chrono = { version = "0.4.23", default-features = false, features = ["clock"] }
+chrono = { version = "0.4.22", default-features = false, features = ["clock"] }
 codemap = "0.1.3"
 codemap-diagnostic = "0.1.1"
-deno_core = "0.174.0"
+deno_core = { git = "https://github.com/exograph/deno.git", branch = "patched" }
```
Comment (ramnivas) [inline Cargo.toml:56]: We should pin the dependency, perhaps by specifying "rev" (https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-git-repositories)

Comment (shadaj) [reply Cargo.toml:56]: It's auto-pinned in the lockfile!

**Pull request merged**